cdispyutils/hmac4/hmac4_auth_utils.py

```diff
@@ -177,6 +177,11 @@ def get_request_date(req):
     for header in [constants.REQUEST_DATE_HEADER, 'date']:
         if header not in req.headers:
             continue
+        try: 
+            date = datetime.datetime.strptime(req.headers[header], '%Y%m%d')
+            break
+        except ValueError:
+            continue
         try:
```

Comment:
can you add some documentation about your changes to the date header? Not very clear to me when looking at the code